method.rb

```diff
@@ -0,0 +1,248 @@
+module Enumerable # rubocop:disable Metrics/ModuleLength
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    output_array = is_a?(Array) ? self : to_a
+    i = 0
+    while i < size
+      yield output_array[i]
+      i += 1
+    end
+    output_array
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    output_array = is_a?(Array) ? self : to_a
+    i = 0
+    while i < size
+      yield output_array[i], i
+      i += 1
+    end
+    output_array
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    output_array = []
+    my_each do |item|
+      output_array << item if (yield item) == true
+    end
+    output_array
+  end
+
+  def parameter_condition?(parameter)
+    parameter.is_a?(Regexp) || parameter.is_a?(Class) || parameter.nil?
+  end
+
+  def my_all?(parameter = nil) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/MethodLength, Metrics/AbcSize
+    status_cumulator = true
+    i = 0
+    bg = block_given?
+    if bg == true && parameter.nil?
+      loop do
+        status_cumulator &&= yield self[i]
+        i += 1
+        break if i == size || status_cumulator == false
+      end
+    elsif bg == true && parameter.is_a?(Regexp)
+      puts 'warning: given block not used'
+      loop do
+        status_cumulator &&= !parameter.match(self[i]).nil?
+        i += 1
+        break if i == size || status_cumulator == false
+      end
+    elsif bg == true && parameter.is_a?(Class)
+      puts 'warning: given block not used'
+      loop do
+        status_cumulator &&= self[i].is_a?(parameter)
+        i += 1
+        break if i == size || status_cumulator == false
+      end
+    elsif bg == true && parameter_condition?(parameter)
+      puts 'warning: given block not used'
+      loop do
+        status_cumulator &&= (self[i] == parameter)
+        i += 1
+        break if i == size || status_cumulator == false
+      end
+    elsif bg == false && parameter.nil?
+      loop do
+        break if i == size || status_cumulator == false
+
+        status_cumulator &&= !self[i].nil?
+        i += 1
+      end
+    elsif bg == false && parameter.is_a?(Regexp)
+      loop do
+        status_cumulator &&= !parameter.match(self[i]).nil?
+        i += 1
+        break if i == size || status_cumulator == false
+      end
+    elsif bg == false && parameter.is_a?(Class)
+      loop do
+        status_cumulator &&= self[i].is_a?(parameter)
+        i += 1
+        break if i == size || status_cumulator == false
+      end
+    else
+      loop do
+        status_cumulator &&= (self[i] == parameter)
+        i += 1
+        break if i == size || status_cumulator == false
+      end
+    end
+    status_cumulator
+  end
+
+  def my_any?(parameter = nil) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/MethodLength, Metrics/AbcSize
```

Comment:
**my_any** when no block or argument is given it should return true if at least one of the collection is not false or nil